core/unix/signal.c

```diff
@@ -3223,8 +3237,16 @@ copy_frame_to_stack(dcontext_t *dcontext, int sig, sigframe_rt_t *frame, byte *s
         info->signal_restorer_retaddr = (app_pc)f_new->pretcode;
 #        endif
 #    endif /* X86 */
-        /* Store the prior mask, for restoring in sigreturn. */
-        convert_rt_mask_to_nonrt(f_new, &info->app_sigblocked);
+        if (info->pre_syscall_app_sigprocmask_valid) {
+            /* Store the prior mask that had been saved pre-syscall, for restoring in
+             * sigreturn.
+             */
+            convert_rt_mask_to_nonrt(f_new, &info->pre_syscall_app_sigprocmask);
+            info->pre_syscall_app_sigprocmask_valid = false;
```

Comment:
?  No, the local prevents having any conditional in the two rt and non-rt cases: they would be the same as TOT but using the local instead of &info->app_sigblocked.

Reply:
Done.